geoips/plugins/modules/readers/generic_satpy.py

```diff
@@ -0,0 +1,134 @@
+values should be a list of channel names that have the same resolution. The name
+of each group will be used as the keys of the output dictionary and the list of
+channels will be grouped together as a Dataset stored as the associated value to
+the key.  So for example the following channel_groups JSON Object could be used
+to read a selection of visible and IR GOES16 data:
+"channel_groups":{"0.5km":["C02"], "2km":["C13", "C14"]}
+
+While Satpy provides a very large number of readers, a very small number of them
+have been used with this plugin.  This plugin should be considered experimental
+and is not guaranteed to work with all of Satpy's readers.
+"""
+
+import logging
+from typing import Any, Dict, List, Optional, Union
+
+import pyresample.geometry
+import satpy
+import xarray as xr
+
+log = logging.getLogger(__name__)
+
+interface = "readers"
+family = "standard"
+name = "generic_satpy"
+
+
+def call(
+    fnames: List[str],
+    source_name: str,
+    platform_name: str,
+    metadata_only: bool = False,
+    chans: Optional[List[str]] = None,
+    area_def: Optional[pyresample.geometry.AreaDefinition] = None,
+    self_register: bool = False,
+    satpy_reader: str = "abi_L1b",
+    channel_groups: Dict[str, List[str]] = {"0.5km": ["C02"]},
+    roi: float = 3000,
+    filter_parameters: Optional[Dict[str, Any]] = None,
+    satpy_reader_kwargs: Optional[Dict[str, Any]] = None,
+    calibration: Union[List[str], str] = "*",
+    resolution: Union[List[float], float] = "*",
+    polarization: Union[List[str], str] = "*",
+    level: Union[List[float], str] = "*",
+    modifiers: Union[List[str], str] = "*",
+    generate: bool = True,
+    unload: bool = True,
+    load_kwargs: Optional[Dict[str, Any]] = None,
+    force_compute: bool = False,
+) -> Dict[str, xr.DataArray]:
+    if platform_name is None:
+        platform_name = satpy_reader
+
+    scene = satpy.Scene(
+        filenames=fnames,
+        reader=satpy_reader,
+        filter_parameters=filter_parameters,
+        reader_kwargs=satpy_reader_kwargs,
+    )
+
+    all_channels = []
+    for channel_names in channel_groups.values():
+        all_channels.extend(channel_names)
+
+    if load_kwargs is None:
+        load_kwargs = {}
+
+    scene.load(
+        all_channels,
+        calibration=calibration,
+        resolution=resolution,
+        polarization=polarization,
+        level=level,
+        modifiers=modifiers,
+        generate=generate,
+        unload=unload,
+        **load_kwargs,
+    )
+
+    output_datasets = {}
+    for group_name, channel_names in channel_groups.items():
+        lons, lats = (None, None)
+        is_first_channel = True
+        data_dict = {}
+        for channel_name in channel_names:
+            channel = scene[channel_name]
+            if is_first_channel:
+                is_first_channel = False
+                lons_np, lats_np = channel.attrs["area"].get_lonlats()
```

Comment:
`get_lonlats()` is pretty inconsistent in what it returns. For `SwathDefinition`s like those created for VIIRS and other LEO/polar instruments, Satpy will probably be providing the swath definition an xarray DataArray but `get_lonlats()` will give you the dask array underneath. For `AreaDefinition`s it would be numpy arrays by default I think. You could pass `chunks=...` to this method and then you'd always get dask arrays and then could compute them together with `lons_np, lats_np = dask.array.compute(lons_dask, lats_dask)` or not compute them if you want GeoIPS to deal with the dask arrays later on.

Reply:
At the moment, I think everything downstream from the reader plugin assumes that it's receiving numpy arrays. (Please correct me if I'm wrong @mindyls and @jsolbrig). So to me it makes sense to make sure that `get_lonlats()` is generating a dask array that is then forced to generate numpy arrays.

Reply:
Correct - at the moment dask arrays passed through GeoIPS would probably be a negative (because they'd end up being computed a bunch of times during sectoring / interpolating).  Computing them once in the reader at this point probably makes sense.  We'd have to implement better support for dask arrays throughout the standard geoips tools.